src/core/data.jl

```diff
@@ -380,52 +451,72 @@ function _clean_status!(network::Dict{String,Any})
 end
 
 
-"Transforms a single network into a multinetwork with several deepcopies of the original network. Indexed from 0."
+"""
+    replicate_restoration_network(sn_data::Dict{String,<:Any}; count::Int=1, global_keys::Set{String}=Set{String}())
+
+Transforms a single network into a multinetwork with several deepcopies of the original network.
+Start with nwid "0" before repairs are conducted, then `count` number of preriods where repairs are performed.
+Add network keys for `repairs`, the number of repairs that are allowed in each period, and `repaired_total`, the cumulative repairs in the network.
+Also add `time_elapsed` based on the number of repairs that occur in the following period.
+"""
 function replicate_restoration_network(sn_data::Dict{String,<:Any}; count::Int=1, global_keys::Set{String}=Set{String}())
     return replicate_restoration_network(sn_data, count, union(global_keys, _PM._pm_global_keys))
 end
 
 
-"Transforms a single network into a multinetwork with several deepcopies of the original network. Indexed from 0."
+"""
+    replicate_restoration_network(sn_data::Dict{String,<:Any}; count::Int=1, global_keys::Set{String}=Set{String}())
+
+Transforms a single network into a multinetwork with several deepcopies of the original network.
+Start with nwid "0" before repairs are conducted, then `count` number of preriods where repairs are performed.
+Add network keys for `repairs`, the number of repairs that are allowed in each period, and `repaired_total`, the cumulative repairs in the network.
+Also add `time_elapsed` based on the number of repairs that occur in the following period.
+"""
+function replicate_restoration_network(sn_data::Dict{String,<:Any}, count::Int; global_keys::Set{String}=Set{String}())
+    return replicate_restoration_network(sn_data, count, union(global_keys, _PM._pm_global_keys))
+end
+
+
+"""
+    replicate_restoration_network(sn_data::Dict{String,<:Any}; count::Int=1, global_keys::Set{String}=Set{String}())
+
+Transforms a single network into a multinetwork with several deepcopies of the original network.
+Start with nwid "0" before repairs are conducted, then `count` number of preriods where repairs are performed.
+Add network keys for `repairs`, the number of repairs that are allowed in each period, and `repaired_total`, the cumulative repairs in the network.
+Also add `time_elapsed` based on the number of repairs that occur in the following period.
+"""
 function replicate_restoration_network(sn_data::Dict{String,<:Any}, count::Int, global_keys::Set{String})
-    pm_sn_data = _PM.get_pm_data(sn_data)
+    pm_sn_data = _PM.get_pm_data(deepcopy(sn_data))
```

Comment:
This will be more performant as `deepcopy(_PM.get_pm_data(sn_data))`.

Reply:
good catch